docs/README.md

```diff
@@ -100,103 +101,96 @@ MobX has only a few core concepts. The following snippets can be tried online us
 
 ### Observable state
 
-<i><a style="color: white; background:green;padding:5px;margin:5px;border-radius:2px" href="https://egghead.io/lessons/javascript-sync-the-ui-with-the-app-state-using-mobx-observable-and-observer-in-react">Egghead.io lesson 1: observable & observer</a></i>
-
 MobX adds observable capabilities to existing data structures like objects, arrays and class instances.
-This can simply be done by annotating your class properties with the [@observable](http://mobxjs.github.io/mobx/refguide/observable-decorator.html) decorator (ES.Next).
+This can simply be done by using `makeObservable` in its constructor to annotate your class properties as `observable`.
 
 ```javascript
-import { observable } from "mobx"
+import { makeObservable, observable, action } from "mobx"
 
 class Todo {
     id = Math.random()
-    @observable title = ""
-    @observable finished = false
+    title = ""
+    finished = false
+
+    constructor() {
+        makeObservable(this, {
```

Comment:
I had considered that myself, but I see some problems with this approach. For one, in the original example 'id' is actually *not* an observable, and "makeAutoObservable" would make it so. I would need to declare there's an exception for that and explain what that does before I even say what it automates.

In addition, in its context the text introduces observable state, then actions, then computed values, step by step. If I say "makeAutoObservable" right from the start, it becomes harder to explain *what* is automated and what the rules are for actions, computed, and exceptions, as you cannot see the distinction so clearly. 

So instead I've added a new section that discusses "makeAutoObservable" below this, to show that this can be abbreviated.

I've guessed what the behavior of 'exceptions' should be. Perhaps it's not a list.

Reply:
Well, I do really think that `makeAutoObservable` will be a major way of using MobX because in most common cases you don't need to be explicit. And as such, it should have a more prominent place doc. Imagine yourself as a newcomer. Wouldn't you like to see the most straightforward approach first and care about edge cases later?

Sure, it needs to be restructured here, perhaps come up with an example that doesn't need exceptions. Then you can just fine follow with current way of revealing other parts.

> If I say "makeAutoObservable" right from the start, it becomes harder to explain _what_ is automated and what the rules

I don't see it as a problem. You don't need to explain in detail _what_ it does, just that it will modify a class ready for MobX consumers.

Reply:
Thought a little about it, I think it is quite important to introduce `makeObservable` _first_, so that we can clearly explain and distinguish the different concepts, and it puts everything quite in the face. Then in the later examples (can be pretty quick, but at least after introducing observable / computed / action trinity first) we can introduce makeAutoObservable and use it in the majority of the cases from there. 

The `toggle` problem is solved now, so I think `makeAutoObservable` now will work in a much bigger majority of the cases.

Reply:
I think I have resolved this now. The first example people see is the one from "Gist of MobX", which is much simpler and doesn't do a conceptual introduction. So I've switched that over to "makeAutoObservable". I've taken the conceptual example and merged it in to "Concepts & Principles", as there was a lot of overlap between them. Here I've used "makeObservable" exclusively as it needs to be clear what's going on, with a brief note about "makeAutoObservable".